File: scratch/test-sqlite.js
const Database = require("better-sqlite3");
const { resolve } = require("path");

function testQuery() {
  const dbPath = resolve("geoai_data/geoai.db"); // Fallback path, might need to adjust
  console.log("DB path:", dbPath);
  
  try {
    const db = new Database(dbPath, { readonly: true });
    
    const sql = `
      SELECT "id", "code", "name", "propertyType", "district"
      FROM "BuildingProperty"
      WHERE "deletedAt" IS NULL AND "propertyType" IN ('cafe', 'coffee_shop', 'restaurant', 'food', 'dining') AND "district" = 'Hải Châu'
      ORDER BY "name" ASC
      LIMIT 15
    `;
    
    console.log("Executing query...");
    const start = Date.now();
    const rows = db.prepare(sql).all();
    console.log(`Query took ${Date.now() - start}ms, returned ${rows.length} rows.`);
    db.close();
  } catch (err) {
    console.error("DB error:", err);
  }
}
testQuery();

File: test_sqlite.py
import sqlite3
import os

db_path = os.path.join(os.path.dirname(__file__), 'geoai_data', 'geoai.db')
print(f"Checking {db_path}")

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    cursor.execute("SELECT COUNT(*) FROM BuildingProperty")
    print(f"BuildingProperty count: {cursor.fetchone()[0]}")
    
    cursor.execute("SELECT COUNT(*) FROM Place")
    print(f"Place count: {cursor.fetchone()[0]}")
    
    cursor.execute("SELECT COUNT(*) FROM User")
    print(f"User count: {cursor.fetchone()[0]}")
except Exception as e:
    print(f"Error: {e}")
